endpoint.go

```diff
@@ -500,8 +500,10 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if e := ep.addToCluster(); e != nil {
-		log.Errorf("Could not update state for endpoint %s into cluster: %v", ep.Name(), e)
+	if sb.ingress {
```

Comment:
I think we should make it the caller's responsibility to call `EnableService/DisableService` even for ingress sandbox. So we probably don't need to do any special case logic here. It's always up to the caller. In case of ingress the caller will call `EnableService` immediately after `Join` and call `DisableService` immediately before `Leave` call.